spec/functions/systemd_escape_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+describe 'systemd::systemd_escape' do
```

Comment:
These tests unit will not work on systems where systemd is not installed (e.g. FreeBSD). While end-users running such os will not use this modules, maintainers at voxpupuli will want to run the test suite locally to ensure they do not break the module with refactoring :wink: 

A more portable way to test this is to mock the behavior of the execution:

```
    allow(Puppet::Util::Execution).to receive(:execute).with('systemd-escape', 'foo').and_return("foo\n")
```

then chek it happened

```
    expect(Puppet::Util::Execution).to have_received(:execute).with('systemd-escape', 'foo')
```